### PR TITLE
Readme docker

### PR DESCRIPTION
**Pull request by ti2ger92**
Links
-----
- Issue: https://github.com/UPchieve/server/issues/359

Description
-----------
- Leverage numbering syntax `1` for each numbered item to have them outnumbered.
- Instruct users to leverage Docker for Mongo and Redis, instead of asdf.
- Add explanation about docker-compose vs individual setup.
- Explicitly state the versions to use (goal of ease of use).
- 

Developer self-review checklist
-------------------------------
- [X] Potentially confusing code has been explained with comments
- [X] No warnings or errors have been introduced; all known error cases have been handled
- [X] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [X] All edge cases have been addressed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,69 +52,78 @@ UPchieve web server
...
+1. If you want to test Twilio voice calling functionality, set the `host` property to `[your public IP address]:3000` (minus the brackets), and configure your router/firewall to allow connections to port 3000 from the Internet. Twilio will need to connect to your system to obtain TwiML instructions.
+1. Run `npm run dev` to start the dev server on `http://localhost:3000`. If you get a [`bcrypt`][bcrypt] compilement error, run `npm rebuild`.
+1. See [the web client repo](https://github.com/UPchieve/web) for client
    installation
-7. (optional) Run `redis-server` and `npm run worker:dev` to start the redis database and dev worker. The dev worker will automatically attempt to connect to your local Redis instance and read jobs from there. Additionally, you can run `ts-node ./scripts/add-cron-jobs.ts` to add all repeatable jobs to the job queue.
+1. (optional) Run `redis-server` and `npm run worker:dev` to start the redis database and dev worker. The dev worker will automatically attempt to connect to your local Redis instance and read jobs from there. Additionally, you can run `ts-node ./scripts/add-cron-jobs.ts` to add all repeatable jobs to the job queue.
```
Comment (jkeat) [inline README.md:122]: Looks like the numbering got messed up?

Comment (jkeat) [inline README.md:122]: Ahh this numbering syntax is neat 👍
